packages/react-dom/index.js

```diff
@@ -1,6 +1,50 @@
 'use strict';
 
+function testMinificationUsedDCE() {
+  if (process.env.NODE_ENV === 'development') {
+    // We expect this method only to be called in production.
+    throw new Error('This is unreachable');
+  }
+  try {
+    // use scoped variable for our initial test, in case
+    // 'top-level' mangling is not enabled.
+    var source = testMinificationUsedDCE.toString();
+    var longVariableName = true;
+    if (longVariableName && source.match(/longVariableName/g).length === 3) {
+      // We are not minified.
+      // This might be a Node environment where DCE is not expected anyway.
+      return;
```

Comment:
Why do we allow this to not be minimized at all?

The other concern was that if shipping works because of this bail out. Then if byte code reversing gets deployed, then it'll skip this and start failing. Without the site doing anything differently.

Reply:
Maybe I don't understand the first question but in Node environment it is expected that code is *not* minimized or altered in any way.

Reply:
This doesn't hit the server entry point. I guess you could still have a `require` on this for `findDOMNode`.

If you run unit tests with react-dom in production, I guess you'd hit this too.

Reply:
Yes, I'm mostly referring to `findDOMNode` case.

I don't think we support running tests in production. For example test renderer throws in entry point if you try that. I guess we could relax that if there are valid use cases.